workitem_blackbox_test.go

```diff
@@ -606,3 +642,433 @@ func TestPagingDefaultAndMaxSize(t *testing.T) {
+	identityRepo := account.NewIdentityRepository(db)
+	identity := account.Identity{
+		FullName: "Test User Integration Random",
+		ImageURL: "http://images.com/42",
+		ID:       newUserUUID,
+	}
+	err := identityRepo.Create(ctx, &identity)
+	if err != nil {
+		fmt.Println("should not happen off.")
+		return nil
+	}
+	return &identity
+}
+
+func TestUpdateWI2(t *testing.T) {
+	resource.Require(t, resource.Database)
+	pub, _ := almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("TestUpdateWI2-Service", almtoken.NewManager(pub, priv), account.TestIdentity)
+	assert.NotNil(t, svc)
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller)
+	payload := app.CreateWorkItemPayload{
+		Type: models.SystemBug,
+		Fields: map[string]interface{}{
+			models.SystemTitle: "Test WI",
+			models.SystemState: "closed"},
+	}
+
+	_, wi := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload)
+
+	defer test.DeleteWorkitemOK(t, svc.Context, svc, controller, wi.ID)
+
+	controller2 := NewWorkitem2Controller(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller2)
+
+	patchPayload := getMinimumRequiredUpdatePayload(wi)
+
+	// update title attribute
+	modifiedTitle := "Is the model updated?"
+	patchPayload.Data.Attributes[models.SystemTitle] = modifiedTitle
+
+	_, updatedWI := test.UpdateWorkitem2OK(t, svc.Context, svc, controller2, wi.ID, patchPayload)
+	require.NotNil(t, updatedWI)
+	assert.Equal(t, updatedWI.Data.Attributes[models.SystemTitle], modifiedTitle)
+
+	// verify self link value
+	if !strings.HasPrefix(*updatedWI.Links.Self, "http://") {
+		assert.Fail(t, fmt.Sprintf("%s is not absolute URL", *updatedWI.Links.Self))
+	}
+	if !strings.HasSuffix(*updatedWI.Links.Self, fmt.Sprintf("/%s", updatedWI.Data.ID)) {
+		assert.Fail(t, fmt.Sprintf("%s is not FETCH URL of the resource", *updatedWI.Links.Self))
+	}
+	patchPayload.Data.Attributes["version"] = strconv.Itoa(updatedWI.Data.Attributes["version"].(int)) // need to do in order to keep object future usage
+
+	// update assignee relationship and verify
+	newUser := createOneRandomUserIdentity(svc.Context, DB)
+	assert.NotNil(t, newUser)
+
+	newUserUUID := newUser.ID.String()
+	patchPayload.Data.Relationships = &app.WorkItemRelationships{}
+	patchPayload.Data.Relationships.Assignee = &app.RelationAssignee{
+		Data: &app.AssigneeData{
+			ID:   &newUserUUID,
+			Type: "some_invalid_type_identities",
+		},
+	}
+
+	test.UpdateWorkitem2BadRequest(t, svc.Context, svc, controller2, wi.ID, patchPayload)
+
+	// update with invalid assignee string (non-UUID)
+	maliciousUUID := "non UUID string"
+	patchPayload.Data.Relationships.Assignee = &app.RelationAssignee{
+		Data: &app.AssigneeData{
+			ID:   &maliciousUUID,
+			Type: "identities",
+		},
+	}
+	test.UpdateWorkitem2BadRequest(t, svc.Context, svc, controller2, wi.ID, patchPayload)
```

Comment:
It would have been nice if each test for a result (e.g. OK or BadRequest) was in its own test function. You can use a test suite to handle common setup and tear down phases for all tests.